pyomo/util/config_domains.py

```diff
@@ -0,0 +1,67 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.common.collections import ComponentSet
+from typing import Sequence
+
+
+class ComponentDataSet:
+    """ComponentDataSet(ctype)
+    Domain validation class that accepts singleton or iterable arguments and
+    compiles them into a ComponentSet, verifying that they are all ComponentDatas
+    of type 'ctype.'
+
+    Parameters
+    ----------
+        ctype: The component type(s) of the list
```

Comment:
We should update this to match usage: `ctype: either a single component type or an iterable of component types`